module/zfs/dsl_crypt.c

```diff
@@ -786,6 +786,13 @@ spa_keystore_load_wkey(const char *dsname, dsl_crypto_params_t *dcp,
 		goto error;
 	}
 
+	/* confirm that dd is the encryption root */
+	ret = dsl_dir_get_encryption_root_ddobj(dd, &rddobj);
+	if (ret != 0 || rddobj != dd->dd_object) {
+		ret = (SET_ERROR(EINVAL));
```

Comment:
no `( )` needed around `SET_ERROR`, without the immediate `return`.  Please remove them, and get the other two instances of this in the function.